nc models/app.model.js

```diff
@@ -45,11 +46,24 @@ exports.allArticles = (sort_by = "created_at", order = "desc") => {
       articles.article_img_url, 
       COUNT(comments.comment_id) AS comment_count
   FROM articles
-  LEFT JOIN comments ON comments.article_id = articles.article_id
-  GROUP BY articles.article_id
-  ORDER BY ${sort_by} ${order} `;
+  LEFT JOIN comments ON comments.article_id = articles.article_id `;
 
-  return db.query(sqlQuery).then(({ rows }) => {
+  if (topic) {
+    sqlQuery += `WHERE articles.topic = $1 `;
+    values.push(topic);
+  }
+
+  sqlQuery += `
+        GROUP BY articles.article_id
+        ORDER BY ${sort_by} ${order}; `;
+
+  return db.query(sqlQuery, values).then(({ rows }) => {
+    if (topic && rows.length === 0) {
```

Comment:
If the topic is an existing topic but had no associated articles, this would always return a 404. This would be a problem as the topic _should_ be found if it exists! Have a think about how you might check if the topic exists and how to respond if does (even if it has no associated articles). The lecture on Tuesday might help!